chia/full_node/full_node_rpc_client.py

```diff
@@ -36,11 +36,13 @@ async def get_blockchain_state(self) -> dict[str, Any]:
             response["blockchain_state"]["peak"] = BlockRecord.from_json_dict(response["blockchain_state"]["peak"])
         return cast(dict[str, Any], response["blockchain_state"])
 
-    async def get_block(self, header_hash: bytes32) -> Optional[FullBlock]:
+    async def get_block(self, header_hash: bytes32) -> FullBlock:
         try:
             response = await self.fetch("get_block", {"header_hash": header_hash.hex()})
-        except Exception:
-            return None
+        except ResponseFailureError as e:
+            if "not found" in str(e):
```

Comment:
this looks suspicious. what if you make a request to a node by hostname and you get "host not found". Is there a good reason to not pass through the `ResponseFailureError`?
This applies to the other places here too.